controllers/datafoundation.go

```diff
@@ -243,6 +258,20 @@ func (r *dataFoundationReconciler) parseSpec(offering *v1alpha1.ManagedFusionOff
 	return nil
 }
 
+func (r *dataFoundationReconciler) reconcileWatchers() error {
```

Comment:
Why is this function named `reconcile`? It does not do any kind of reconciliation.  Also, why do we need the function at all? this code should be embedded at the start of reconcilePhases.